src/bin/miri.rs

```diff
@@ -166,11 +172,25 @@ fn main() {
     rustc_driver::install_ice_hook();
 
     // If the environment asks us to actually be rustc, then do that.
-    if env::var_os("MIRI_BE_RUSTC").is_some() {
+    if let Some(crate_kind) = env::var_os("MIRI_BE_RUSTC") {
         rustc_driver::init_rustc_env_logger();
         // We cannot use `rustc_driver::main` as we need to adjust the CLI arguments.
         let mut callbacks = rustc_driver::TimePassesCallbacks::default();
-        run_compiler(env::args().collect(), &mut callbacks)
+        run_compiler(
+            env::args().collect(),
+            &mut callbacks,
+            // Don't insert `MIRI_DEFAULT_ARGS`, in particular, `--cfg=miri`, if we are building a
+            // "host" crate. That may cause procedural macros (and probably build scripts) to depend
+            // on Miri-only symbols, such as `miri_resolve_frame`:
+            // https://github.com/rust-lang/miri/issues/1760
+            if crate_kind == "target" {
+                true
+            } else if crate_kind == "host" {
+                false
+            } else {
+                panic!("invalid `MIRI_BE_RUSTC` value: {:?}", crate_kind)
+            }
```

Comment:
Can you move this logic into `let insert_default_args = ...`? Currently it is not very clear what those booleans values mean.